summingbird-online/src/main/scala/com/twitter/summingbird/online/executor/AsyncBase.scala

```diff
@@ -107,13 +136,15 @@ abstract class AsyncBase[I, O, S, D, RC](maxWaitingFutures: MaxWaitingFutures, m
     if (pendingFuturesCount > maxWaitingFuturesCount) {
       // Too many futures waiting, let's clear.
       outstandingFutures.dequeueAll(_.isDefined)
-      val toForce = outstandingFutures.trimTo(maxWaitingFuturesCount).toIndexedSeq
```

Comment:
Yes, it would have timed out with MaxFutureWaitTime. Now, that is not true, so the users need to force timeouts on their futures, otherwise they could last quite long.

I think that will be okay, but I bet some folks will see weird corner cases. Be prepared.

That said, I'm optimistic this PR can really improve throughput quite a lot, so that's exciting.